### PR TITLE
AAP-25873: Dirty fix for 400 Bad Request from WCA requests

### PR DESCRIPTION
**Pull request by manstis**
Jira Issue: https://issues.redhat.com/browse/AAP-25873

## Description
Quick fix to support improved exception handling for playbook generation/explanation.

## Testing
I've added unit tests for the handled scenarios.

### Steps to test
1. Pull down the PR
2. Unit tests pass.

### Scenarios tested
Unit tests.

## Production deployment
- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

**Comment by manstis** [comment]
@goneri @robinbobbitt I can write a ton of unit tests tomorrow, if you'd prefer the dirty fix to be semi-clean.

File: ansible_ai_connect/ai/api/views.py [hunk 1]
```diff
@@ -770,10 +841,83 @@ def post(self, request) -> Response:
                 "format": "plaintext",
                 "generationId": generation_id,
             }
+
+        except ModelTimeoutError as e:
```
Comment (robinbobbitt) [inline ansible_ai_connect/ai/api/views.py:845]: We don't currently raise a ModelTimeoutError for playbook gen or explanation. Having a timeout on these new calls looks like a gap currently. Shall we remove this handling in both places for now for clarity?

File: ansible_ai_connect/ai/api/views.py [hunk 2]
```diff
@@ -770,10 +841,83 @@ def post(self, request) -> Response:
...
+
+        except WcaNoDefaultModelId as e:
+            exception = e
+            logger.exception(
+                "A default WCA Model ID was expected but not found for "
+                f"playbook generation {generation_id}"
+            )
+            raise WcaNoDefaultModelIdException(cause=e)
+
+        except WcaSuggestionIdCorrelationFailure as e:
```
Comment (robinbobbitt) [inline ansible_ai_connect/ai/api/views.py:887]: We're not yet setting the x-request-id header on playbook gen or explanation. Shall we remove this handling in both places for clarity?

File: ansible_ai_connect/ai/api/views.py [hunk 3]
```diff
@@ -770,10 +841,83 @@ def post(self, request) -> Response:
...
+        except ModelTimeoutError as e:
+            exception = e
+            logger.warn(
+                f"model timed out after {settings.ANSIBLE_AI_MODEL_MESH_API_TIMEOUT} seconds"
+                f" for playbook generation {generation_id}"
+            )
+            raise ModelTimeoutException(cause=e)
+
+        except WcaBadRequest as e:
+            exception = e
```
Comment (robinbobbitt) [inline ansible_ai_connect/ai/api/views.py:854]: The main goal in pushing this out the quick-and-dirty way was to get some extra logging in the case where we're getting 400 errors from WCA. The current handling doesn't log the response body anywhere. Shall we add some extra handling like we have here? https://github.com/ansible/ansible-ai-connect-service/blob/main/ansible_ai_connect/ai/api/model_client/wca_utils.py#L161-L163

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
Thanks for jumping on this, @manstis ! Just a couple quick updates please. I'm ok on the code coverage front given the super short intended lifespan of this change.

File: ansible_ai_connect/ai/api/views.py (hunk 3 above)
Comment (manstis) [reply ansible_ai_connect/ai/api/views.py:832]: Indeed.

I now call the chain of response handlers that perform the exact logging you link.

https://github.com/ansible/ansible-ai-connect-service/pull/1157/files#diff-052b0c3ff4c4ad8cbf30bbdaa0f88e9157b5e33a94c6ea85d38257f7010b576eR495-R496

https://github.com/ansible/ansible-ai-connect-service/pull/1157/files#diff-052b0c3ff4c4ad8cbf30bbdaa0f88e9157b5e33a94c6ea85d38257f7010b576eR525-R526

File: ansible_ai_connect/ai/api/views.py (hunk 2 above)
Comment (manstis) [reply ansible_ai_connect/ai/api/views.py]: I've removed `WcaSuggestionIdCorrelationFailure` handling.

File: ansible_ai_connect/ai/api/views.py (hunk 3 above)
Comment (robinbobbitt) [reply ansible_ai_connect/ai/api/views.py:832]: 400s aren't getting logged though, just 404s. You can confirm by trying to generate a playbook with just punctuation as input. You'll see the response body isn't logged anywhere.

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
Please see my response on 400 response body being logged. Thanks!

File: ansible_ai_connect/ai/api/views.py (hunk 3 above)
Comment (manstis) [reply ansible_ai_connect/ai/api/views.py:832]: @robinbobbitt I now [log ](https://github.com/ansible/ansible-ai-connect-service/pull/1157/files#diff-1705f63f7b89989770c6c1a60b68c2857dd8729e8d2fee5024eaef42dab3a089R48-R56)the details if _any_ exception thrown by our response handling; or would you rather just log it for 404's and 400's?

Comment (robinbobbitt) [reply ansible_ai_connect/ai/api/views.py:832]: Nice! That seems good to me. Thank you.

**Review by robinbobbitt — APPROVED** [review]

**Pull request merged**